### PR TITLE
Change Logger.Log specification to "should not actually panic or exit"

### PR DESCRIPTION
**Pull request by jcorbin**
Towards #166

### CODE REVIEW

**Comment by jcorbin** [comment]
Updated #161 to use this, to resolve its prior note on Fatal and Panic

File: logger.go [hunk 1]
```diff
@@ -50,6 +50,11 @@ type Logger interface {
 
 	// Log a message at the given level. Messages include any context that's
 	// accumulated on the logger, as well as any fields added at the log site.
+	//
+	// Calling Panic should panic(), calling Fatal() should terminate the
```
Comment (akshayjshah) [inline logger.go]: nit: "...should panic() *and* calling Fatal()..."

File: logger_test.go [hunk 2]
```diff
@@ -184,19 +184,52 @@ func TestJSONLoggerLog(t *testing.T) {
 		logger.Log(DebugLevel, "foo")
 		assert.Equal(t, `{"level":"debug","msg":"foo"}`, buf.Stripped(), "Unexpected output from Log.")
 	})
+}
 
-	withJSONLogger(t, nil, func(logger Logger, buf *testBuffer) {
-		assert.Panics(t, func() { logger.Log(PanicLevel, "foo") }, "Expected logging at Panic level to panic.")
-		assert.Equal(t, `{"level":"panic","msg":"foo"}`, buf.Stripped(), "Unexpected output from panic-level Log.")
-	})
+func TestJSONLoggerLog_panic(t *testing.T) {
```
Comment (akshayjshah) [inline logger_test.go]: I don't think that we're using this underscore convention in the rest of the test suite; purely for consistency, can we call these `TestJSONLoggerLogPanic` and `...Fatal`?

Comment (akshayjshah) [inline logger_test.go]: Whoops, I lied - we're using this in the bark wrapper code. Mea culpa, do as you wish.

**Comment by akshayjshah** [comment]
lgtm, merge away.

**Pull request merged**